internal/controller/nodedisruption_controller_test.go

```diff
@@ -85,89 +114,203 @@ var _ = Describe("NodeDisruption controller", func() {
 			Expect(k8sClient.Create(ctx, node3)).Should(Succeed())
 		})
 
-		AfterEach(func() {
-			clearAllNodeDisruptionRessources()
-		})
+		Context("With reconciler with default config", Ordered, func() {
+			BeforeAll(func() {
+				cancel_fn = start_reconciler_with_config(NodeDisruptionReconcilerConfig{
+					RejectEmptyNodeDisruption: false,
+					RetryInterval:             time.Second * 1,
+				})
+			})
 
-		Context("When there are no budgets in the cluster", func() {
-			It("It grants the node disruption", func() {
-				By("By creating a new NodeDisruption")
-				disruption := &nodedisruptionv1alpha1.NodeDisruption{
-					TypeMeta: metav1.TypeMeta{
-						APIVersion: "nodedisruption.criteo.com/v1alpha1",
-						Kind:       "NodeDisruption",
-					},
-					ObjectMeta: metav1.ObjectMeta{
-						Name:      NDName,
-						Namespace: NDNamespace,
-					},
-					Spec: nodedisruptionv1alpha1.NodeDisruptionSpec{
-						NodeSelector: metav1.LabelSelector{MatchLabels: node_labels1},
-					},
-				}
-				Expect(k8sClient.Create(ctx, disruption.DeepCopy())).Should(Succeed())
+			AfterAll(func() {
+				cancel_fn()
+			})
 
-				NDLookupKey := types.NamespacedName{Name: NDName, Namespace: NDNamespace}
-				createdDisruption := &nodedisruptionv1alpha1.NodeDisruption{}
+			defer AfterEach(func() {
+				clearAllNodeDisruptionRessources()
+			})
 
-				By("By checking the NodeDisruption is being granted")
-				Eventually(func() nodedisruptionv1alpha1.NodeDisruptionState {
-					err := k8sClient.Get(ctx, NDLookupKey, createdDisruption)
-					if err != nil {
-						panic("should be able to get")
+			Context("When there are no budgets in the cluster", func() {
+				It("It grants the node disruption", func() {
```

Comment:
Note: no need to repeat the context word in text e.g `It("grants...` `By("checking...`